packages/router/router.ts

```diff
@@ -2309,7 +2345,11 @@ export function unstable_createStaticHandler(
     if (matchesToLoad.length === 0) {
       return {
         matches,
-        loaderData: {},
+        // Add a null for all matched routes for proper revalidation on the client
+        loaderData: matches.reduce(
+          (acc, m) => Object.assign(acc, { [m.route.id]: null }),
+          {}
+        ),
```

Comment:
Ensure proper hydration by always sending null values for non-executed route loaders on the server.  This is needed to handle differences where they may be no loader on the server,  but there _is_ a loader on the client (for route JS chunk fetching).  The client gets confused on initial load and decides "this route is active but I've got no hydrated loader data so I need to do an initial fetch).  It also impacts subsequent routing since it doesn't see this route as a revalidation.

Another approach would be for the client-side router to treat _any_ `hydrationData` as complete and simply add `null` there for any active routes without `hydrationData`.